js-pkg/server/src/get_binary.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs/promises')
+const path = require('path')
+
+function getSuffix(os_type, os_arch) {
+  if (os_type === 'Windows_NT' && os_arch === 'x64') return 'win-x64.exe.gz'
+  if (os_type === 'Linux' && os_arch === 'x64') return 'linux-x64.gz'
+  //we rely on rosetta to provide emulation on aarch64 for now
+  if (os_type === 'Darwin') return 'macos-x64.gz'
+
+  throw new Error(`Unsupported platform: ${os_type} ${os_arch}`)
+}
+
+function binaryUrl(version, os_type, os_arch) {
+  const suffix = getSuffix(os_type, os_arch)
+  const url = `https://github.com/drifting-in-space/y-sweet/releases/download/v${version}/y-sweet-server-${suffix}`
+  return url
+}
+
+async function downloadFile(url, filename_or_folder = '.') {
+  const zlib = require('zlib')
+  const { Readable } = require('stream')
+  const { finished } = require('stream/promises')
+  const res = await fetch(url)
+  if (res.status === 404) {
+    throw new Error('no such binary')
```

Comment:
I suggest:

```javascript
if (res.status === 404) {
    throw new Error(`Tried to download ${url} but the file was not found. It may have been removed.`)
} else if (res.status !== 200) {
    throw new Error(`Error downloading ${url}: server returned ${res.status}`)
}
```

Reply:
:+1: